File: recursion/easy/inorder_traversal.py
class BinaryTree:
    def __init__(self, data, left = None, right = None):
        self.data = data
        self.left = left
        self.right = right

def inorderTraversal(root):
    #ここから書きましょう
    def _inorderTraversal(root, arr):
        if root is not None:
            _inorderTraversal(root.left, arr)
            arr.append(root.data)
            _inorderTraversal(root.right,arr)

    arr = []
    _inorderTraversal(root, arr)
    return arr

File: leetcode/easy/inorder_traversal.ts
import { TreeNode } from "../common/common";

function inorderTraversal(root: TreeNode | null): number[] {
  function _inorderTraversal(root: TreeNode | null, arr: number[]): void {
    if (root != null) {
      _inorderTraversal(root.left, arr);
      arr.push(root.val);
      _inorderTraversal(root.right, arr);
    }
  }

  let arr = [];
  _inorderTraversal(root, arr);

  return arr;
}
